esi_ui/api/esi_api.py

```diff
@@ -0,0 +1,33 @@
+from django.conf import settings
+
+from esi import connection
+
+#from horizon.utils.memoized import memoized  # noqa
+
+import openstack.config
+
+
+OS_CLOUD = "openstack"
+
+
+#@memoized
+def esiclient():
+    config = openstack.config.get_cloud_region(cloud=OS_CLOUD)
+    return connection.ESIConnection(config=config)
```

Comment:
We definitely don't want to hardcode the OpenStack connection. The openstack-dashboard webapp will live on an ESI controller, and hardcoding like this means that every user will see the hardcoded user's view. Horizon (and the ironic-ui) must grab the credentials some other way; you should probably copy that.

Reply:
Are we able to use environment variables? If not, I'll try to see if I can get it to work with [https://github.com/openstack/openstacksdk/blob/6328a8fb2f34601400f2014fc24ca4bf9ee340a4/openstack/connection.py#L408](keystone session).

Reply:
Environment variables have a similar issues. Using a keystone session is the way to go; the `ironic-ui`passes in a token (https://github.com/openstack/ironic-ui/blob/master/ironic_ui/api/ironic.py#L45) and somehow generates a session (https://opendev.org/openstack/python-ironicclient/src/branch/master/ironicclient/client.py#L69)

Reply:
The alternative - which would also be fine - would be to update `python-esileapclient` to have the same capability to generate a client that all the other services apparently have.

Reply:
I found out that I can get a keystone token in esi_rest_api.py in the variable `request.user.token.id`. Currently trying to figure out how this can be incorporated with esisdk.

I could, however, make a curl request to the endpoint with the token in a header, if that sounds promising.

Reply:
Yeah, I'm guessing there must be a way somehow to generate a session with a token. I don't know how off the top of my head, though; openstack code sometimes feels a bit... obfuscated. I'll poke around a bit.

Reply:
Try taking the variables that the ironic-ui uses at https://github.com/openstack/ironic-ui/blob/master/ironic_ui/api/ironic.py#L45C1-L50C44 and passing them into ESIConnection. A lot of times those variables are just passed on and on and on until they reach some underlying layer; I wouldn't be entirely unsurprised if that just kinda worked.

Reply:
This worked for me:

```
DEFAULT_OPENSTACK_KEYSTONE_URL = 'http://127.0.0.1/identity/v3'


#@memoized
def esiclient(token):
    auth_url = getattr(settings, 'OPENSTACK_KEYSTONE_URL', DEFAULT_OPENSTACK_KEYSTONE_URL)
    region = config.get_cloud_region(load_yaml_config=False,
                                     load_envvars=False,
                                     auth_type='token',
                                     token=token,
                                     auth_url=auth_url)
    session = region.get_session()
    return connection.ESIConnection(session=session)
```

where token parameter is `response.user.token.id`